networkx/algorithms/link_analysis/hits_alg.py

```diff
@@ -82,8 +82,8 @@ def hits(G,max_iter=100,tol=1.0e-8,nstart=None,normalized=True):
         s=1.0/sum(h.values())
         for k in h:
             h[k]*=s
-    i=0
-    while True: # power iteration: make up to max_iter iterations
+    i=1
+    while i<=max_iter:  # power iteration: make up to max_iter iterations
```

Comment:
This is needlessly complicated; just do `for i in range(max_iter):` then use `max_iter` in the error message below.